Feature/ Search-Input for FilterBar

isacoder opened the pull request:
### Requirements

closes #147 - FilterBar SearchInput

### Implementation

I used the theming for compact input since it look more closer to the zeppelin values. Also there was not an example for non placeholder text, so took the value from the theme. 

However I saw that there were values in typography for SearchInput for the Topbar in the theme, but is very different.
Not sure if this one should have a new theme value. 

I wanted this search input to be able to use in the filter bar but also in the Dropdown component, so I have added the customization.


<img width="541" alt="Screen Shot 2021-08-10 at 18 54 51" src="https://user-images.githubusercontent.com/10409078/128847037-1bab72c0-099d-4e47-9025-37b778ad5d24.png">

Added in dimensions Strong as weight for Icons because heavy looked thiner than the design however it looks a little bit chopped :(. Please advice in this point, should I keep it heavy or do I need to update the icon with more space in the icons repo?

### Screenshoots



<img width="690" alt="Screen Shot 2021-08-10 at 18 57 29" src="https://user-images.githubusercontent.com/10409078/128847462-9b907680-5a14-4e58-9901-5007edf6df93.png">

<img width="447" alt="Screen Shot 2021-08-10 at 18 58 09" src="https://user-images.githubusercontent.com/10409078/128847530-693338b3-09d2-4cb9-ae01-a845bdc408db.png">

atomworks commented on src/themes/common.tsx:
```diff
@@ -8,7 +8,8 @@ export const dimensions = {
     weights: {
       light: 1,
       regular: 1.5,
-      heavy: 2
+      heavy: 2,
+      strong: 3,
```
The weights I have in Sketch are 1, 1.5, 3 and 5 so we definitely need an extra level but it those heavier weights are rarely used so if we're needing them in this feature we need to look at if there's a problem in the icon workflow somewhere. 

Originally the icons should have been setup to scale in the browser in the same way that Sketch handles it. This is part of the reason we have a script to parse icons to add them to the kit. Does the weighting on this feature have the same issue if you choose different icons or is it just an issue with this icon?

isacoder replied:
About the icon weights

<img width="1130" alt="Screen Shot 2021-08-18 at 17 10 42" src="https://user-images.githubusercontent.com/10409078/129862364-90bc53cb-1c84-4dab-b80e-f7fcc932c1ca.png">

Well most of them have this problem and some do not upscale that good 🙈 

Zone
<img width="191" alt="Screen Shot 2021-08-18 at 17 15 39" src="https://user-images.githubusercontent.com/10409078/129863068-0a4efb59-2d1f-4910-a172-559fb4047132.png">

Retry Job
<img width="125" alt="Screen Shot 2021-08-18 at 17 17 02" src="https://user-images.githubusercontent.com/10409078/129863246-4d0bd094-0360-4b60-b295-b2325864fe40.png">

Temperature
<img width="135" alt="Screen Shot 2021-08-18 at 17 17 24" src="https://user-images.githubusercontent.com/10409078/129863296-7b9a4726-96af-4cdf-b3ed-aa8d005028d6.png">

Usage
<img width="142" alt="Screen Shot 2021-08-18 at 17 18 16" src="https://user-images.githubusercontent.com/10409078/129863407-0f5e1d84-9687-4f74-baeb-ef201ca9b822.png">

Critical
<img width="216" alt="Screen Shot 2021-08-18 at 17 14 32" src="https://user-images.githubusercontent.com/10409078/129862914-ba3c6459-9741-4db1-8b11-cf5f2d1fe9a3.png">

Probably we shouldn't use this weight for now..

atomworks reviewed (changes requested):
This looks good although I think the naming of the component needs a change. This is an input that gets used in forms but I think components in the `Form` section should be focused on forms that are used in pages for things like settings, etc. 

So this is more of a utility input used for specific UI elements and should rarely get used outside of other UI kit elements. Let's try and work out a name to better reflect this. This might be better to live as some kind of atom, maybe in global? Or something new? We can chat on Slack if you have some ideas!

isacoder commented:
After the Slack conversation I have updated the SearchInput to BasicSearchInput and moved it to Misc.

I'll be waiting on the retro of the Icons to follow up with any requested changes.

atomworks reviewed (commented):
After our discussion I know there'll be some changes to separate out the icon issue to its own PR. Other than that though, I think we're good.

Let me know when you've done what you need to and I'll do one last check really quickly and get this approved.
